packages/api/lambdas/create-reconciliation-report.js

```diff
@@ -639,13 +651,17 @@ async function createReconciliationReport(recReportParams) {
         bucketReport.onlyInDynamoDb
       );
 
-      Object.keys(bucketReport.okCountByGranule).forEach((granuleId) => {
-        const currentGranuleCount = report.filesInCumulus.okCountByGranule[granuleId];
-        const bucketGranuleCount = bucketReport.okCountByGranule[granuleId];
+      if (linkingFilesToGranules(recReportParams.reportType)) {
+        Object.keys(bucketReport.okCountByGranule).forEach((granuleId) => {
+          const currentGranuleCount = report.filesInCumulus.okCountByGranule[granuleId];
+          const bucketGranuleCount = bucketReport.okCountByGranule[granuleId];
 
-        report.filesInCumulus.okCountByGranule[granuleId] = (currentGranuleCount || 0)
-          + bucketGranuleCount;
-      });
+          report.filesInCumulus.okCountByGranule[granuleId] = (currentGranuleCount || 0)
+            + bucketGranuleCount;
+        });
+      } else {
+        delete report.filesInCumulus.okCountByGranule;
```

Comment:
was "okCountByGranule" ever even added if the report type wasn't GNF? Why would we have to delete it?

Reply:
Yes, it's initialized as an empty set.  We could leave it out here, but then it would exist in the final report as {}

Reply:
https://github.com/nasa/cumulus/pull/2055/files/fabbcb0e63d8b109425e246b7ff8648a454f429a#diff-61b4ede5087a45c0d1191b106bc92b405c40e436ec41bf30bd9b86ef5bb2d634R156

Reply:
https://github.com/nasa/cumulus/blob/mhs/CUMULUS-2351/api-options-to-limit-reconciliation-reports/packages/api/lambdas/create-reconciliation-report.js#L156